tools/e2e-commons/env/prerequisites.js

```diff
@@ -80,26 +80,22 @@ export function prerequisitesBuilder( page ) {
  * @param {page}    page            - Playwright page instance.
  */
 async function buildPrerequisites( state, page ) {
-	const functions = {
-		plugins: () => ensurePluginsState( state.plugins ),
-		loggedIn: () => ensureUserIsLoggedIn( page ),
-		wpComLoggedIn: () => ensureWpComUserIsLoggedIn( page ),
-		connected: () => ensureConnectedState( state.connected ),
-		plan: () => ensurePlan( state.plan, page ),
-		modules: () => ensureModulesState( state.modules ),
-		clean: () => ensureCleanState( state.clean ),
-	};
+	const functions = [
+		[ 'clean', () => ensureCleanState( state.clean ) ],
+		[ 'plugins', () => ensurePluginsState( state.plugins ) ],
+		[ 'loggedIn', () => ensureUserIsLoggedIn( page ) ],
+		[ 'wpComLoggedIn', () => ensureWpComUserIsLoggedIn( page ) ],
+		[ 'connected', () => ensureConnectedState( state.connected ) ],
+		[ 'plan', () => ensurePlan( state.plan, page ) ],
+		[ 'modules', () => ensureModulesState( state.modules ) ],
+	];
 
 	logger.prerequisites( JSON.stringify( state, null, 2 ) );
 
-	for ( const option of Object.keys( state ) ) {
-		if ( state[ option ] !== undefined ) {
-			if ( functions[ option ] ) {
-				logger.prerequisites( `Ensuring '${ option }' prerequisite state` );
-				await functions[ option ]();
-			} else {
-				throw Error( `Unknown state "${ option }: ${ state[ option ] }"!` );
```

Comment:
Decided to remove this error handling. With this change the function will always be defined, but any unknown state keys will be ignored.

Could add something to check that the state keys are correct, but it seems overkill to me.